tests/main/snap-userd/task.yaml

```diff
@@ -0,0 +1,119 @@
+summary: Ensure snap userd allows opening a URL via xdg-open
+
+systems:
+    # Not supposed to work on Ubuntu Core systems as we don't have
+    # a user session environment there
+    - -ubuntu-core-*
+
+environment:
+    DISPLAY: :0
+
+restore: |
+    . "$TESTSLIB/pkgdb.sh"
+    rm -f dbus.env
+    if [[ "$SPREAD_SYSTEM" == ubuntu-14.04-* ]]; then
+        stop test-snap-userd || true
+        rm -f /etc/init/test-snap-userd.conf
+    else
+        systemctl stop --signal=KILL test-snap-userd.scope || true
+    fi
+    umount -f /usr/bin/xdg-open || true
+    umount -f /snap/core/current/usr/bin/xdg-open || true
+    distro_purge_package dbus-x11 xdg-utils
+
+execute: |
+    . "$TESTSLIB/pkgdb.sh"
+    . "$TESTSLIB/dirs.sh"
+
+    # Install necessary pacakges to get dbus-launch helper
+    distro_install_package dbus-x11 xdg-utils
+
+    dbus-launch > dbus.env
+    export $(cat dbus.env | xargs)
+
+    if [[ "$SPREAD_SYSTEM" == ubuntu-14.04-* ]]; then
+        cat << EOF > /etc/init/test-snap-userd.conf
+    env DISPLAY="$DISPLAY"
+    env DBUS_SESSION_BUS_ADDRESS="$DBUS_SESSION_BUS_ADDRESS"
+    env DBUS_SESSION_BUS_PID="$DBUS_SESSION_BUS_PID"
+    kill timeout 5
+    exec /usr/bin/snap userd
+    EOF
+
+        initctl reload-configuration
+        start test-snap-userd
+        # This is a bit nasty but the best we can do at the moment as we don't
+        # have something like `busctl status io.snapcraft.SafeLauncher` on 14.04
+        list_dbus_names() {
+            dbus-send --session            \
+                --dest=org.freedesktop.DBus \
+                --type=method_call          \
+                --print-reply               \
+                /org/freedesktop/DBus       \
+                org.freedesktop.DBus.ListNames
+        }
+        while ! list_dbus_names | MATCH io.snapcraft.SafeLauncher ; do
+            sleep .1
+        done
+    else
+        systemd-run \
+            --scope \
+            --unit=test-snap-userd \
+            --no-block \
+            --setenv=DISPLAY="$DISPLAY" \
+            --setenv=DBUS_SESSION_BUS_ADDRESS="$DBUS_SESSION_BUS_ADDRESS" \
+            --setenv=DBUS_SESSION_BUS_PID="$DBUS_SESSION_BUS_PID" \
+            /bin/sh -c '/usr/bin/snap userd &'
+
+        # Wait until the service is up and running
+        while ! systemctl is-active test-snap-userd.scope ; do
+            sleep .1
```

Comment:
The ping seems to be java specific, but I did something in a similar spirit in the branch now.